### PR TITLE
Add example for Self provisioned AWS S3 or S3-compatible storage credential creation

### PR DESCRIPTION
**Pull request by ramya-bangera**
1. **What this PR does / why we need it**:
   Add example for Self provisioned AWS S3 or S3-compatible storage credential creation

2. **Which issue(s) this PR fixes**

       Internal Jira ticket

3. **Additional information**
This change adds example for creatingSelf provisioned AWS S3 or S3-compatible storage credential creation for data protection

4. **Special notes for your reviewer**

### CODE REVIEW

**Review by ankitsny — DISMISSED** [review]

File: docs/data-sources/kubernetes_secret.md [hunk 1]
```diff
@@ -30,14 +30,14 @@ Only one scope per resource is allowed.
 ```terraform
 # Read Tanzu Mission Control kubernetes secret : fetch namespace details
 data "tanzu-mission-control_kubernetes_secret" "read_secret" {
-  name                    = "tf-secret" # Required
-  namespace_name          = "tf-secret-namespace-name" # Required 
+  name           = "tf-secret"                # Required
+  namespace_name = "tf-secret-namespace-name" # Required
```
Comment (shobha2626) [inline docs/data-sources/kubernetes_secret.md:34]: Is the script formatted?

File: docs/resources/credential.md [hunk 2]
```diff
@@ -172,6 +172,39 @@ resource "tanzu-mission-control_credential" "tmc_provisioned_aws_s3_cred" {
...
+      key_value {
+        type = "OPAQUE_SECRET_TYPE"
+        data = {
+          "aws_access_key_id"          = "abcd="
+          "aws_secret_access_key"      = "xyz=="
+        }
+      }
+    }
+  }
+  ready_wait_timeout = "1m" // Wait time for credential create operations to finish (default: 3m).
```
Comment (shobha2626) [inline docs/resources/credential.md:204]: 1m is lesser compared to the default time, I assume the crew creation takes more than 1m.

**Review by shobha2626 — CHANGES_REQUESTED** [review]

File: docs/resources/credential.md (hunk 2 above)
Comment (ramya-bangera) [reply docs/resources/credential.md:204]: It was good enough when I tested, anyways I have increased it to 5m now

File: docs/data-sources/kubernetes_secret.md (hunk 1 above)
Comment (ramya-bangera) [reply docs/data-sources/kubernetes_secret.md:34]: Yes formatted now

File: docs/resources/credential.md (hunk 2 above)
Comment (shobha2626) [reply docs/resources/credential.md:204]: The change is not reflecting here

Comment (ramya-bangera) [reply docs/resources/credential.md:204]: Please refresh, it is already updated

Comment (shobha2626) [reply docs/resources/credential.md:204]: Please regenerate the doc, to reflect the changes

**Review by shobha2626 — APPROVED** [review]
LGTM

**Review by ankitsny — APPROVED** [review]

**Pull request merged**